Gems/Multiplayer/Code/Source/Components/SimplePlayerSpawnerComponent.cpp

```diff
@@ -72,36 +74,74 @@ namespace Multiplayer
         incompatible.push_back(AZ_CRC_CE("MultiplayerSpawnerService"));
     }
 
-    NetworkEntityHandle SimplePlayerSpawnerComponent::OnPlayerJoin(
-        [[maybe_unused]] uint64_t userId, [[maybe_unused]] const MultiplayerAgentDatum& agentDatum)
+    AZ::Transform SimplePlayerSpawnerComponent::RoundRobinNextSpawnPoint()
     {
-        const PrefabEntityId prefabEntityId(AZ::Name(m_playerSpawnable.m_spawnableAsset.GetHint().c_str()));
-        AZ::Transform transform = AZ::Transform::Identity();
+        const AZ::Transform nextSpawnPoint = GetNextSpawnPoint();
+        m_spawnIndex = ++m_spawnIndex % m_spawnPoints.size();
+        return nextSpawnPoint;
+    }
+
+    AZ::Transform SimplePlayerSpawnerComponent::GetNextSpawnPoint() const
+    {
+        if (!GetNextSpawnPointIndex().IsSuccess())
+        {
+            return AZ::Transform::Identity();
+        }
+
+        const AZ::EntityId spawnPointEntityId = m_spawnPoints[m_spawnIndex];
+        const AZ::Entity* spawnPointEntity = AZ::Interface<AZ::ComponentApplicationRequests>::Get()->FindEntity(spawnPointEntityId);
+        if (spawnPointEntity != nullptr)
+        {
+            return spawnPointEntity->GetTransform()->GetWorldTM();
+        }
```

Comment:
Just curious, is this better in some way than using the Transform bus?

Reply:
Nope, I just derped >.<
Updated to just use the ebus. Thank you, good catch.